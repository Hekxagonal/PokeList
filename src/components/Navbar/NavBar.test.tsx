import { fireEvent, screen } from '@testing-library/react';
import renderTheme from '../../global/theme/render-theme';

import Component from './'

describe('<NavBar />', () => {
    const defaultValues = {
    }
    it('should be change theme color', () => {
        renderTheme(<Component {...defaultValues}/>)


        const container = screen.getByTestId('container')
        expect(container).toHaveStyleRule('background', 'rgba(168,0,0,1)')
        const ThemeIcon = screen.getByTestId('themeButton')
        fireEvent(ThemeIcon, new MouseEvent('click', {}))
        expect(container).toHaveStyleRule('background', 'rgba(168,0,0,1)')
    })
    it('should be a title', () => {
        renderTheme(<Component {...defaultValues}/>)


        const title = screen.getByRole('heading', { name: 'PokeList'})
        expect(title).toBeInTheDocument()
    })

    it('should be match snapshot', () => {
        const view = renderTheme(<Component {...defaultValues}/>)

        expect(view).toMatchSnapshot()
    })
})