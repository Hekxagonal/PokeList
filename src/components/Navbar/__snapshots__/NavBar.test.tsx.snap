// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should be match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background: rgba(168,0,0,1);
  color: rgba(255,255,255,1);
  position: fixed;
  z-index: 10;
  width: 100%;
  height: 50px;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
}

.c1 {
  margin-left: 10px;
}

.c4 {
  margin-right: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: inline;
  color: white;
  font-size: 30px;
}

.c3 {
  color: black;
  font-size: 15px;
  font-weight: bold;
  color: white;
}

.c5 {
  color: white;
}

<body>
    <div>
      <div
        class="c0"
        data-testid="container"
      >
        <div
          class="c1"
        >
          <h1
            class="c2"
          >
            PokeList 
          </h1>
          <span
            class="c3"
          >
            by Flavio : )
          </span>
        </div>
        <div
          class="c4"
        >
          <svg
            class="c5"
            data-testid="themeButton"
            fill="currentColor"
            height="35px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="35px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 2H7c-1.103 0-2 .897-2 2v3c0 1.103.897 2 2 2h11c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zM7 7V4h11l.002 3H7z"
            />
            <path
              d="M13 15v-2c0-1.103-.897-2-2-2H4V5c-1.103 0-2 .897-2 2v4c0 1.103.897 2 2 2h7v2a1 1 0 0 0-1 1v5a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1v-5a1 1 0 0 0-1-1z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  background: rgba(168,0,0,1);
  color: rgba(255,255,255,1);
  position: fixed;
  z-index: 10;
  width: 100%;
  height: 50px;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
}

.c1 {
  margin-left: 10px;
}

.c4 {
  margin-right: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: inline;
  color: white;
  font-size: 30px;
}

.c3 {
  color: black;
  font-size: 15px;
  font-weight: bold;
  color: white;
}

.c5 {
  color: white;
}

<div>
    <div
      class="c0"
      data-testid="container"
    >
      <div
        class="c1"
      >
        <h1
          class="c2"
        >
          PokeList 
        </h1>
        <span
          class="c3"
        >
          by Flavio : )
        </span>
      </div>
      <div
        class="c4"
      >
        <svg
          class="c5"
          data-testid="themeButton"
          fill="currentColor"
          height="35px"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="35px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 2H7c-1.103 0-2 .897-2 2v3c0 1.103.897 2 2 2h11c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zM7 7V4h11l.002 3H7z"
          />
          <path
            d="M13 15v-2c0-1.103-.897-2-2-2H4V5c-1.103 0-2 .897-2 2v4c0 1.103.897 2 2 2h7v2a1 1 0 0 0-1 1v5a1 1 0 0 0 1 1h2a1 1 0 0 0 1-1v-5a1 1 0 0 0-1-1z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
